### PR TITLE
Hits rerun

### PR DESCRIPTION
**Pull request by smotherh**
Several improvements to the code, with a focus on compatibility to the Lori Allen dataset and Region Search functionality.

### CODE REVIEW

File: analysis/analysis_utils.py [hunk 1]
```diff
@@ -227,8 +228,11 @@ def process_results(self,search,image_params,res_filepath,likelihood_level):
         print("Processing Results")
         print('---------------------------------------')
         while likelihood_limit is False:
+            print('Starting pooling...')
             pool = mp.Pool(processes=16)
-            results = search.get_results(res_num,chunk_size)
+            print('Getting results...')
+            if results is None:
+                results = search.get_results(res_num,chunk_size)
```
Comment (jbkalmbach) [inline analysis/analysis_utils.py]: This breaks the loop. You only ever get the results in the first chunk. This needs to be fixed before merging.

File: analysis/analysis_utils.py [hunk 2]
```diff
@@ -227,7 +227,9 @@ def process_results(self,search,image_params,res_filepath,likelihood_level):
         print("Processing Results")
         print('---------------------------------------')
         while likelihood_limit is False:
+            print('Starting pooling...')
             pool = mp.Pool(processes=16)
+            print('Getting results...')
             results = search.get_results(res_num,chunk_size)
```
Comment (jbkalmbach) [inline analysis/analysis_utils.py]: Now if you put something other than `None` in results, it never gets used. Did you want that option to still work?

Comment (smotherh) [reply analysis/analysis_utils.py]: No. I think that was originally an attempt to get a single process_results class for grid search and region search. I eventually gave up and coded a separate process_region_results() function.

File: search/src/KBMOSearch.cpp [hunk 3]
```diff
@@ -876,7 +903,32 @@ std::vector<RawImage> KBMOSearch::phiStamps(trajectory& t, int radius)
...
 	return createStamps(t, radius, imgs);
 }
-
+std::vector<float> KBMOSearch::psiCurves(trajectory& t)
+{
+    /*Generate a psi lightcurve for further analysis
+     *  INPUT-
+     *    trajectory& t - The trajectory along which to find the lightcurve
+     *  OUTPUT-
+     *    std::vector<double> - A vector of the lightcurve values
```
Comment (PWhiddy) [inline search/src/KBMOSearch.cpp]: Change 'double' to 'float'

File: search/src/KBMOSearch.cpp [hunk 4]
```diff
@@ -876,7 +903,32 @@ std::vector<RawImage> KBMOSearch::phiStamps(trajectory& t, int radius)
...
+	for (auto& im : psiImages) imgs.push_back(&im);
+	return createCurves(t, imgs);
+}
+std::vector<float> KBMOSearch::phiCurves(trajectory& t)
+{
+    /*Generate a phi lightcurve for further analysis
+     *  INPUT-
+     *    trajectory& t - The trajectory along which to find the lightcurve
+     *  OUTPUT-
+     *    std::vector<double> - A vector of the lightcurve values
```
Comment (PWhiddy) [inline search/src/KBMOSearch.cpp]: double -> float

File: search/src/KBMOSearch.cpp [hunk 5]
```diff
@@ -798,6 +798,33 @@ std::vector<RawImage> KBMOSearch::createStamps(trajectory t, int radius, std::ve
...
+std::vector<float> KBMOSearch::createCurves(trajectory t, std::vector<RawImage*> imgs)
+{
+    /*Create a lightcurve from an image along a trajectory
+     *
+     *  INPUT-
+     *    trajectory t - The trajectory along which to compute the lightcurve
+     *    std::vector<RawImage*> imgs - The image from which to compute the
+     *      trajectory. Most likely a psiImage or a phiImage.
+     *   Output-
+     *     std::vector<double> lightcurve - The computed trajectory
```
Comment (PWhiddy) [inline search/src/KBMOSearch.cpp]: double -> float

File: analysis/analysis_utils.py [hunk 6]
```diff
@@ -194,14 +193,14 @@ def load_images(self, im_filepath, time_file, psf_val=1.4, mjd_lims=None):
...
+        #stack.apply_master_mask(master_flags, 2)
 
-        stack.grow_mask()
-        stack.grow_mask()
+        #stack.grow_mask()
+        #stack.grow_mask()
         
         # This applies a mask to pixels with more than 120 counts
-        #stack.apply_mask_threshold(120.)
+        stack.apply_mask_threshold(120.)
```
Comment (jbkalmbach) [inline analysis/analysis_utils.py]: Should this be turned back on?

File: analysis/analysis_utils.py [hunk 7]
```diff
@@ -7,6 +7,7 @@
 import csv
 import astropy.coordinates as astroCoords
 import astropy.units as u
+import pdb
```
Comment (jbkalmbach) [inline analysis/analysis_utils.py]: Should we remove pdb imports before moving things to master?

File: analysis/analysis_utils.py (hunk 6 above)
Comment (smotherh) [reply analysis/analysis_utils.py]: I think so. Without using bitmasks, having a pixel threshold eliminates a lot of the ultra-high likelihood objects (e.g. 100+).

File: analysis/analysis_utils.py (hunk 7 above)
Comment (smotherh) [reply analysis/analysis_utils.py]: Probably. Done.

**Pull request merged**